src/library/utils.lua

```diff
@@ -352,6 +353,9 @@ function utils.show_notes_dialog(caption, width, height)
 
         width = width or 500
         height = height or 350
+        if (parent and not parent.ExecuteModal) then
+            parent = nil -- probably not a valid window object
+        end
```

Comment:
Passing a `parent` that is not a window is a program bug and should therefore throw. You have two choices:

- let the call to `ExecuteModal` throw if it's the wrong type
- use `assert`, e.g.:

```lua
assert(parent and not parent.ExecuteModal, "argument 4 must be nil or an instance of FCResourceWindow")
```

Reply:
`assert` throws the error where it is called. For something like argument validation, I usually find it more helpful to have the error be thrown a level higher, to where the current function was called. If you do want to do that, you can follow this proforma:

```lua
if argument_is_bad then
    error("error message", 2)
end
```

Reply:
Very good. Amended. This now throws a helpful error message back at the right part of the calling script.